### ISSUE TITLE
bug: Error no asdf when trying to run sass (No preset version installed for command sass)

### ISSUE THREAD

**Issue by Victorcorcos** [asdf-vm/asdf#1656]
### Describe the Bug

# Steps to reproduce

## 1. Install `sass`

* https://blog.teamtreehouse.com/install-sass-macos-homebrew

```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
brew install sass/sass/sass
```

## 2. Run `sass`

```
sass --version
```

### Steps to Reproduce

# Steps to reproduce

## 1. Install `sass`

* https://blog.teamtreehouse.com/install-sass-macos-homebrew

```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
brew install sass/sass/sass
```

## 2. Run `sass`

```
sass --version
```

## Output

<img width="1155" alt="Screen Shot 2023-09-29 at 19 36 34" src="https://github.com/asdf-vm/asdf/assets/7637806/7a849121-577f-44be-abc5-e589202267af">

```
No preset version installed for command sass
Please install a version by running one of the following:

asdf install ruby 3.0.6

or add one of the following versions in your config file at /Users/victorcosta/Desktop/repositories/dpms-front-norway/.tool-versions
ruby 2.6.8
```

1. When I execute the recommended command: `asdf install ruby 3.0.6`, it just says: `ruby 3.0.6 is already installed`
2. I also tried to do `asdf reshim ruby`, but it doesn't work.

Does someone know how to fix it?

Thanks!

### Expected Behaviour

The `sass` should be executed properly.

## Output

<img width="1155" alt="Screen Shot 2023-09-29 at 19 36 34" src="https://github.com/asdf-vm/asdf/assets/7637806/7a849121-577f-44be-abc5-e589202267af">

```
No preset version installed for command sass
Please install a version by running one of the following:

asdf install ruby 3.0.6

or add one of the following versions in your config file at /Users/victorcosta/Desktop/repositories/dpms-front-norway/.tool-versions
ruby 2.6.8
```

1. When I execute the recommended command: `asdf install ruby 3.0.6`, it just says: `ruby 3.0.6 is already installed`
2. I also tried to do `asdf reshim ruby`, but it doesn't work.

Does someone know how to fix it?

Thanks!

### Environment

```shell
$ asdf info
OS:
Darwin Victors-MacBook-Pro.local 20.6.0 Darwin Kernel Version 20.6.0: Thu Mar  9 20:39:26 PST 2023; root:xnu-7195.141.49.700.6~1/RELEASE_X86_64 x86_64

SHELL:
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin20)
Copyright (C) 2007 Free Software Foundation, Inc.

BASH VERSION:
3.2.57(1)-release

ASDF VERSION:
v0.12.0

ASDF INTERNAL VARIABLES:
ASDF_DEFAULT_TOOL_VERSIONS_FILENAME=.tool-versions
ASDF_DATA_DIR=/Users/victorcosta/.asdf
ASDF_DIR=/usr/local/opt/asdf/libexec
ASDF_CONFIG_FILE=/Users/victorcosta/.asdfrc

ASDF INSTALLED PLUGINS:
python                       https://github.com/tuvistavie/asdf-python.git master 8505457
ruby                         https://github.com/asdf-vm/asdf-ruby.git master 8cb9721
```

<p align="center">
  <img src="https://i.imgur.com/16RW3pd.png" alt="Example"/>
</p>
```


### asdf plugins affected (if relevant)

_No response_

**Comment by Victorcorcos** [comment]
I want to run the global `sass`, installed by `brew install sass/sass/sass`. 

* https://sass-lang.com/install/

However, it looks like `asdf` is interferring that and trying to run a `sass` related to a specific `ruby` version.

How can I run the global `sass` instead of the `asdf` `sass`?

Thanks

**Comment by felixwatts** [comment]
I have the same problem

**Comment by Birdie0** [comment]
Had similar issue with sass. Since brew init command usually gets added to `~/.zprofile` (`~/.bash_profile` if using bash) and asdf one goes to `~/.zshrc` (`~/.bashrc`), asdf shim directory will appear first in path, shadowing brew binaries.

You might want to move brew init line to `~/.zshrc` (`~/.bashrc`) under asdf init or vice versa.

**Comment by joaocv3** [comment]
This is what did the trick for me:

`rm ~/.asdf/shims/sass`

**Comment by allisonphillips** [comment]
> [!NOTE]
> TL;DR, your personal development environment is configured to prioritize looking in asdf's shims before it looks for definitions in your homebrew install, and you still have some value shimmed for that command that was installed for a different language version than the language version in the context where you're running the command.

If using npm, try doing a global install of sass from the root of the project where sass isn't being recognized because the package (or one of its dependencies) don't appear to have a version installed for the version of node being used in the context you ran the command from.

If you're trying to use the homebrew tap you'll want to modify your terminal's configuration file (.zshrc/.bashrc) to look at homebrew first (homebrew needs to be before asdf shims in your `$PATH`)... your computer needs to be configured to search homebrew's executables before asdf's shims if you want it to be using homebrew's executables. You may then need to run additional commands to remove your computer's preexisting symlinks for sass if some lookup table on your computer continues mapping the command to asdf. Deleting any asdf shims for sass will also probably fix it, but you're still likely to encounter the same problem with a different symptom if your computer (either now or in the future) gets another hit on a sass executable somewhere else in your search path before it gets to homebrew or recreates the shim (for instance, by doing a global install of sass via npm). Alternatively, opening the shim executable in a text editor and modifying the exec statement to run the command using homebrew instead of asdf is also likely to work, though I'd discourage doing so.

## I don't see anything to really 'fix' here...
It's not asdf's responsibility to manipulate your PATH or magically know that you want to switch to using homebrew to resolve a package name that asdf has a shimmed value for, and attempting to do so would _create_ much more confusing buglike behavior. This is an edge case that someone will only experience if they are trying to run an executable AND the executable's name already matches a shim in asdf AND the computer's search path is configured to search asdf's shims ahead of the directories where the newly-preferred executable can be found.

That said, asdf's error messaging of `No preset version installed for command NAME`, which appears to occur when the name is shimmed but there is no executable for the current context's language plugin, is vague to the point of being misleading. Because it's followed with a suggestion to install the language plugin version, it reads like it's telling you to install the plugin language version you already have installed; something like `No installation of command 'NAME' could be found for PLUGIN_NAME PLUGIN_VERSION, perhaps 'NAME' needs to be installed?` might read more clearly to the user.

---

#### Anecdote
I ran into this with sass-migrator, which I had initially installed via `npm install -g sass-migrator`. The global install puts it in `~/.asdf/installs/nodejs/VERSION_A/lib/modules/sass-migrator`, where VERSION_A is the version of node being used in the terminal window I initially ran the global install from.

When I went to run sass-migrator on a different repository that uses a different version of node we'll call VERSION_B, it spat out this error:

```
No preset version installed for command sass-migrator
Please install a version by running one of the following:

asdf install nodejs VERSION_B
```

asdf recognized sass-migrator as a shimmed command because of the unix executable in .asdf/shims:
```bash
#!/usr/bin/env bash
# asdf-plugin: nodejs VERSION_A
exec /opt/homebrew/opt/asdf/libexec/bin/asdf exec "sass-migrator" "$@" # asdf_allow: ' asdf '
``` 

But it couldn't map to the executable I was trying to run because I'd never previously installed it in a context that was using nodejs VERSION_B. After doing a global install of the package in a context using nodejs VERSION_B, the shim executable looks like this:

```bash
#!/usr/bin/env bash
# asdf-plugin: nodejs VERSION_A
# asdf-plugin: nodejs VERSION_B
exec /opt/homebrew/opt/asdf/libexec/bin/asdf exec "sass-migrator" "$@" # asdf_allow: ' asdf '
``` 

...and the error no longer occurs when running the command.

Hopefully this helps some of the people commenting here better understand this is not a bug and learn some things to help independently debug this kind of problem in the future.

**Comment by RobLoach** [comment]
Not sure if this helps anyone, but libsass is deprecated in light of [dart-sass](https://github.com/sass/dart-sass), which can be installed with Node.js and [asdf-nodejs](https://github.com/asdf-vm/asdf-nodejs):
```
npm install -g sass
```

**Comment by jthegedus** [comment]
Please see this part of the documentation:

https://asdf-vm.com/manage/versions.html#fallback-to-system-version

>To use the system version of tool <name> instead of an asdf managed version you can set the version for the tool to system.

EG:

```
# $HOME/.tool-versions

golang 1.23.2
node 22.9.0
sass system

# other tools...
```

It should then fallback to the system resolution mechanism and pick up the Homebrew installed `sass`.

**Comment by jthegedus** [comment]
Please open an new issue if this continues to fail once adopting the system fallback mechanism.
